conans/test/unittests/test_client/test_graph/test_range_resolver.py

```diff
@@ -0,0 +1,50 @@
+import unittest
+from conans.client.graph.range_resolver import _parse_versionexpr
+from conans.test.utils.tools import TestBufferConanOutput
+from conans.errors import ConanException
+
+
+class ParseVersionExpr(unittest.TestCase):
+    def test_backwards_compatibility(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3, 3.2", output), ("2.3 3.2", True, False))
+        self.assertEqual(_parse_versionexpr("2.3, <=3.2", output), ("2.3 <=3.2", True, False))
+
+    def test_standard_semver(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("*", output), ("*", True, False))
+        self.assertEqual(_parse_versionexpr("", output), ("", True, False))  # Defaults to '*'
+        self.assertEqual(_parse_versionexpr("~1", output), ("~1", True, False))
+        self.assertEqual(_parse_versionexpr("~1.2.3-beta.2", output), ("~1.2.3-beta.2", True, False))
+        self.assertEqual(_parse_versionexpr("^0.0", output), ("^0.0", True, False))
+        self.assertEqual(_parse_versionexpr("1.2.3 - 2.3.4", output), ("1.2.3 - 2.3.4", True, False))
+        self.assertEqual(_parse_versionexpr(">=1.2.3 <1.(2+1).0", output), (">=1.2.3 <1.(2+1).0", True, False))
```

Comment:
I think this syntax is not valid in semver although  it is allowed in the regex. I think this comes from the **explanation** of the tilde ranges here: https://docs.npmjs.com/misc/semver#tilde-ranges-123-12-1

I have done some tests with this script and it is not working as expected:

```
from semver import SemVer, max_satisfying

filter = ">= 1.(1+1).0"

versions = [SemVer("1.0.0", loose=False), SemVer("1.2.0", loose=False)]
result = max_satisfying(versions, filter)
print("Result:", result)
```
The output is ``Result: None``. Whereas if I change the filter to `` ">= 1.2.0"`` it outputs the right result ``Result: 1.2.0``

Reply:
Yes @danimtb, you are right. The syntax stands for `~1.2.3`. This seems to be bug in the semver library. Let's issue there a bug report.

Reply:
What now, remove it? Leave round brackets inside regex?

Reply:
Raise a `ConanException` if it is not valid syntax (remove `(` and `)` from regex).